### PR TITLE
ContextualMenu - allow dismiss to be prevented on itemClick.

### LINKED ISSUE #3140: Allow clicking commandbar context menu item without dismissing the menu
### Bug Report

- __Package version(s)__: 5.0.1
- __Browser and OS versions__: (fill this out if relevant)

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)

Requested priority: (Blocking, High, Normal, Low) 
High

Products/sites affected: (if applicable)
Owa mail command bar Move to menu

#### Describe the issue:

Overriding onDismiss for a context menu item of command bar should allow an option to not dismiss the menu. Right now it calls onDismiss callback but still dismisses the menu by calling setState.
We have a requirement to add menu items in owa mail command bar's Move to menu for -
1) New Folder - Clicking this will replace the menu item with a text box where user can enter new folder name. On enter it should dismiss the menu.
2) More Folders - Clicking on this menu item should not dismiss the Move to menu. It will replace the folder list with full folder list in the same menu (not open a submenu).

    CommandBar.prototype._onContextMenuDismiss = function (ev) {
        if (!ev || !ev.relatedTarget || !this.refs.commandSurface.contains(ev.relatedTarget)) {
            var contextualMenuProps = this.state.contextualMenuProps;
            if (contextualMenuProps && contextualMenuProps.onDismiss) {
                this.state.contextualMenuProps.onDismiss(ev);
            }
            this.setState({
                expandedMenuItemKey: undefined,
                contextualMenuProps: undefined,
                contextualMenuTarget: undefined
            });
        }
        else {
            ev.stopPropagation();
            ev.preventDefault();
        }
    };

#### Actual behavior:

Clicking on an item always dismisses the context menu

#### Expected behavior:

Do not dismiss the menu onDismiss callback

### PR DESCRIPTION
**Pull request by micahgodbolt**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #3140
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by micahgodbolt** [comment]
well drat, the contextual menu exposes the dismiss function, but the button, in the command bar, doesn't. Will it be possible to get a reference to the contextual menu and close it manually if needed, or will that need to be at the button level?

**Comment by christiango** [comment]
Can you provide more details on the scenario and how this change works?

**Comment by micahgodbolt** [comment]
Right now contextual menu clicks automatically close the menu. This would allow someone to write an onClick function that prevented that default action. maybe they are opening up a modal, or some other piece of navigation, and they don't want the menu to close.

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx [hunk 1]
```diff
@@ -680,7 +680,7 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
       this.props.onItemClick(ev, item);
     }
 
-    this.dismiss(ev, true);
+    !ev.defaultPrevented && this.dismiss(ev, true);
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: Can we add an example of this somewhere on the demo site so people can see how to use this?

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: Hope to get to this tomorrow.

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: A great example from the linked issue is to click on a contextual menu and have it change to an input

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: or having a "see more" button that you can click and see more links in menu

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx]: context

**Pull request merged**